### ISSUE TITLE
Avoid eval

### ISSUE THREAD

**Issue by jathri** [caolan/async#1403]
Hello!
I would like to ask if you can avoid using eval. I am getting this error from Firefox browser console:
Content Security Policy: The page’s settings blocked the loading of a resource at self (“script-src moz-extension://...”). Source: call to eval() or related function blocked by CSP.
async.js:206

**Comment by aearly** [comment]
If you can find a workaround for our use of `eval` I would be very interested:

https://github.com/caolan/async/blob/master/lib/internal/wrapAsync.js#L6-L15

**Comment by ex1st** [comment]
@aearly why not to change:

```javascript
function isAsync(fn) {
    return supportsSymbol && fn[Symbol.toStringTag] === 'AsyncFunction';
}
```
to
```javascript
function isAsync(fn) {
   return fn.constructor.name === 'AsyncFunction';
}
```

**Comment by aearly** [comment]
Released in v2.4.0!

**Comment by jathri** [comment]
Thank you.
